stdlib/influxdata/influxdb/alerts/alerts.flux

```diff
@@ -31,3 +32,29 @@ logs = (start, stop=now(), fn) =>
 deadman = (t, tables=<-) => tables
     |> max(column: "_time")
     |> map(fn: (r) => ( {r with dead: r._time < t} ))
+
+// levels describing the result of a check
+levelOK = 0
+levelInfo = 1
+levelWarn = 2
+levelCrit = 3
+levelUnknown = -1
+
+// Check performs a check against its input using the given ok, info, warn and crit functions
+// and writes the result to a system bucket.
+check = (tables=<-, data, messageFn, crit, warn, info, ok) =>
+    tables
+        |> experimental.set(o: data.tags)
+        |> experimental.group(mode: "extend", columns: experimental.objectKeys(o: data.tags))
+        |> map(fn: (r) => ({r with
+            _check_id:  data._check_id,
+            _level:
+                if crit(r: r) then levelCrit
+                else if warn(r: r) then levelWarn
+                else if info(r: r) then levelInfo
+                else if ok(r: r) then levelOK
+                else levelUnknown
+        }))
+        |> map(fn: (r) => ({r with _message: messageFn(r: r, data: data)}))
+        |> experimental.group(mode: "extend", columns: ["_check_id", "_level"])
```

Comment:
This looks good, of the above we need to add `_check_name` `_type` and `_source_timestamp`

Reply:
Won't `_source_timestamp` create high cardinality?  Maybe you mean `_source_measurement`?